docs/pages/example/data-join.html

```diff
@@ -72,30 +72,35 @@
     // https://www.census.gov/geo/maps-data/data/cbf/cbf_state.html
     map.addSource("states", {
         type: "vector",
-        url: "mapbox://mapbox.us_census_states_2015"
+        url: "mapbox://mapbox.10asuvo4",
     });
 
-    var expression = ["match", ["get", "STATEFP"]];
-
-    // Calculate color for each state based on the unemployment rate
-    data.forEach(function(row) {
-        var green = (row["unemployment"] / maxValue) * 255;
-        var color = "rgba(" + 0 + ", " + green + ", " + 0 + ", 1)";
-        expression.push(row["STATE_ID"], color);
-    });
-
-    // Last value is the default, used where there is no data
-    expression.push("rgba(0,0,0,0)");
-
-    // Add layer from the vector tile source with data-driven style
     map.addLayer({
         "id": "states-join",
         "type": "fill",
         "source": "states",
         "source-layer": "states",
         "paint": {
-            "fill-color": expression
+            "fill-color": ["rgba",
+                0,
+                [ "*", [ "/", ["to-number", ["feature-state", "unemployment"]], maxValue], 255],
+                0,
+                0.9]
         }
     }, 'waterway-label');
+
+    function setStates() {
+        data.forEach(function(row) {
+            map.setFeatureState({source: 'states', sourceLayer: 'states', id: row["STATE_ID"]}, row);
```

Comment:
I think a comment here describing what this does would be useful – I don't think it would necessarily be obvious to a user that you're associating the JSON data to the features on the map.